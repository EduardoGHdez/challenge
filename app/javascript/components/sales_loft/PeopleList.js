import React from "react"

function PeopleList({ people }) {
  const peopleRows = people.map(({ id, emailAddress, title, displayName }) =>
    <tr key={id.toString()}>
      <td><span>Name</span>{ displayName }</td>
      <td><span>Email</span>{ emailAddress }</td>
      <td><span>Job Title</span>{ title }</td>
    </tr>
  );

  return (
    <section>
      <h1><b>People</b></h1>
      <table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Email</th>
            <th>Job Title</th>
          </tr>
        </thead>
        <tbody>
         { peopleRows }
        </tbody>
      </table>
    </section>
  );
}

export default PeopleList
