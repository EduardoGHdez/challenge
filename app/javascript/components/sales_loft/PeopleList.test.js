import React from "react";
import renderer from 'react-test-renderer';

import PeopleList from "./PeopleList";

const mockedPeople = [
  { id: 1, emailAddress: "homer.simpson@gmail.com", displayName: "Homer Simpson", title: "Sr. Software Engineer" },
  { id: 2, emailAddress: "bart.simpson@gmail.com", displayName: "Bart Simpson", title: "Software Engineer" },
  { id: 3, emailAddress: "lisa.simpson@gmail.com", displayName: "Lisa Simpson", title: "Tech Lead" }
];

describe("<PeopleList />", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PeopleList people={mockedPeople} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
