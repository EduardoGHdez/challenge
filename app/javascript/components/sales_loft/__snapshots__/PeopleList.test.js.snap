// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeopleList /> renders correctly 1`] = `
<section>
  <h1>
    <b>
      People
    </b>
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Job Title
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span>
            Name
          </span>
          Homer Simpson
        </td>
        <td>
          <span>
            Email
          </span>
          homer.simpson@gmail.com
        </td>
        <td>
          <span>
            Job Title
          </span>
          Sr. Software Engineer
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Name
          </span>
          Bart Simpson
        </td>
        <td>
          <span>
            Email
          </span>
          bart.simpson@gmail.com
        </td>
        <td>
          <span>
            Job Title
          </span>
          Software Engineer
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Name
          </span>
          Lisa Simpson
        </td>
        <td>
          <span>
            Email
          </span>
          lisa.simpson@gmail.com
        </td>
        <td>
          <span>
            Job Title
          </span>
          Tech Lead
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
